use e2epod.WaitTimeoutForPodReadyInNamespace directly

tanjunchen opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
 
 /kind cleanup
 

**What this PR does / why we need it**:

WaitForPodReady is simply wrapper functions for e2epod package, and they made an invalid dependency to sub e2e framework from the core framework.

So we can use e2epod.WaitTimeoutForPodReadyInNamespace  to remove invalid dependency.
The main purpose of this pr is to handle the framework core package dependency subpackage pod.
For subpackage e2e/framework/pod we can consider encapsulating common code after remove e2epod in framework.go.

/cc @oomichi
for review . Thank you very much.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
ref https://github.com/kubernetes/kubernetes/issues/81245

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs
None
```

tanjunchen commented:
/assign @alejandrox1

mattjmcnaughton commented on test/e2e/framework/framework.go:
```diff
@@ -520,11 +520,6 @@ func (f *Framework) WaitForPodNotFound(podName string, timeout time.Duration) er
 	return e2epod.WaitForPodNotFoundInNamespace(f.ClientSet, podName, f.Namespace.Name, timeout)
 }
 
-// WaitForPodReady waits for the pod to flip to ready in the namespace.
```
From the failed tests, I think we may have missed some users of `WaitForPodReady`.

tanjunchen replied:
you are right.Thanks,i will update it.

mattjmcnaughton commented on test/e2e/framework/framework.go:
```diff
@@ -520,11 +520,6 @@ func (f *Framework) WaitForPodNotFound(podName string, timeout time.Duration) er
 	return e2epod.WaitForPodNotFoundInNamespace(f.ClientSet, podName, f.Namespace.Name, timeout)
```
Can you more about why we are getting rid of the `WaitForPodReady` wrapper but not, for example, the `WaitForPodNotFound` wrapper?

tanjunchen replied:
thanks for  your review . WaitForPod*()  in test/e2e/framework/framework.go  should be replaced to remove the dependency of core framework for sub pod . see  issue https://github.com/kubernetes/kubernetes/issues/81245

mattjmcnaughton commented on test/e2e/common/container_probe.go:
```diff
@@ -63,7 +63,7 @@ var _ = framework.KubeDescribe("Probing container", func() {
 		containerName := "test-webserver"
 		p := podClient.Create(testWebServerPodSpec(probe.withInitialDelay().build(), nil, containerName, 80))
-		f.WaitForPodReady(p.Name)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, p.Name, f.Namespace.Name, framework.PodStartTimeout))
```
Did we intend to add the `ExpectNoError` here?

tanjunchen replied:
> Thanks for all your work to clean up these tests.
> 
> I'm having trouble seeing where this code actually removes a dependency - very possible I'm missing something... or perhaps its a precursor to a diff that will remove some imports?

see https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L54  and

tanjunchen replied:
> Did we intend to add the `ExpectNoError` here?

This may not be related to this pr, but the BB function will return an error, and you see other similar code that calls this BB function to handle errors. I propose to add error handling here. What do you think?

oomichi replied:
I know this kind thing depends on each person.
But I prefer to see a single PR should do a single thing, in this case this PR should just replace functions without doing anything because that makes unnecessary discussion like this.
It is better to add error handling with another PR if necessary.

tanjunchen replied:
@alejandrox1  Thank you for your comment, as @oomichi  said, this pr is currently for the purpose of eliminating dependencies, I can handle this error  in another pr. 😄

mattjmcnaughton reviewed (commented):
Thanks for all your work to clean up  these tests.

I'm having trouble seeing where this code actually removes a dependency - very possible I'm missing something... or perhaps its a precursor to a diff that will remove some imports?

alejandrox1 reviewed (commented):
Thank you @tanjunchen for looking into this one!
/kind cleanup
/priority backlog
/lgtm
for now, i think some more work needs to get tests green.

Also for general info, @mattjmcnaughton you asked a very good question
> I'm having trouble seeing where this code actually removes a dependency - very possible I'm missing something... or perhaps its a precursor to a diff that will remove some imports?

Testing commons, a subproject of SIG testing, has been looking into cleaning up the e2e test framework (the lib in `k/k/test/e2e/framework/`).
One of the goals set was to decouple the framework itself from any helper libraries that were used to execute e2e tests.
That way the framework can just help with test setup and framework subpackages (i.e., e2epod in k/k/test/e2e/framework/pod/ ) can help with pod things.
On that front right now the framework functions WaitForPodReady just wrap calls to e2epod. To reduce the number of code dependencies the goal is to rewrite the tests in order to use e2epod directly. WHich should hopefully make it easier as well to maintain the e2e framework.

tanjunchen commented on test/e2e/framework/network/utils.go:
```diff
@@ -664,7 +664,7 @@ func (config *NetworkingTestConfig) createNetProxyPods(podName string, selector
 	runningPods := make([]*v1.Pod, 0, len(nodes))
 	for _, p := range createdPods {
-		framework.ExpectNoError(config.f.WaitForPodReady(p.Name))
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(config.f.ClientSet, p.Name, config.f.Namespace.Name, framework.PodStartTimeout))
```
i will do it later.:smile:

mattjmcnaughton reviewed (approved):
Gotcha, the incremental approach sounds good to me! Could you please just update the pull request description, or the commit message, so that its clear that there are multiple steps we intend to take? Otherwise, its slightly confusing as a reviewer. Thanks :)

/lgtm

tanjunchen committed:
WaitForPodReady is simply wrapper functions for e2epod package,
and they made an invalid dependency to sub e2e framework from the core framework.

So we can use e2epod.WaitTimeoutForPodReadyInNamespace to remove invalid dependency.

The main purpose of this pr is to handle the framework core package dependency subpackage pod.

tanjunchen commented:
> Gotcha, the incremental approach sounds good to me! Could you please just update the pull request description, or the commit message, so that its clear that there are multiple steps we intend to take? Otherwise, its slightly confusing as a reviewer. Thanks :)
> 
> /lgtm

sure , i add some message in commits

alejandrox1 reviewed (commented):
/approve
/lgtm
